### PR TITLE
marketplace produce card

### PR DESCRIPTION
**Pull request by he-is-james**
produce cards designed to look like the figma
calls data from the airtable and displays them as produce cards on markeplace
started sending data to produce card, need more props for produce detail screen

### CODE REVIEW

File: src/Marketplace/MarketplaceScreen.js [hunk 1]
```diff
@@ -32,11 +46,51 @@ const styles = StyleSheet.create({
...
+  },
 });
 
 export default function MarketplaceScreen({ navigation }) {
+  const [produceList, setProduceList] = useState([]);
+
+  const getProduce = () => {
+    const list = [];
+    base('Produce').select({
+      maxRecords: 20,
```
Comment (hwang635) [inline src/Marketplace/MarketplaceScreen.js]: How come you're limiting the number of records?

File: src/Marketplace/MarketplaceScreen.js [hunk 2]
```diff
@@ -32,11 +46,51 @@ const styles = StyleSheet.create({
...
+      navigation={navigation}
+      image={carrotImage}
+      name={produce.Name}
+      price={produce.Price}
+      unit={produce.Unit}
+    />
+  ));
+
+  useEffect(() => {
+    getProduce();
```
Comment (hwang635) [inline src/Marketplace/MarketplaceScreen.js:115]: This looks good! How were you able to fix the issue of not loading upon mount?

File: src/Marketplace/ProduceCard.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+function ProduceCard({
+  navigation, image, name, price, unit,
+}) {
+  const onPressCard = () => {
+    navigation.navigate('ProduceDetails', {
+      image, name, price, unit,
+    });
+  };
+  return (
+    <TouchableOpacity style={styles.container} onPress={onPressCard}>
```
Comment (hwang635) [inline src/Marketplace/ProduceCard.js:78]: using TouchableOpacity is a good idea!

File: src/Marketplace/ProduceCard.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        <Icon style={styles.favoriteIcon} name="heart" size={15} />
+        <Image style={styles.image} source={image} />
+        <Text style={styles.name}>{name}</Text>
+        <View style={styles.bottom}>
+          <Text style={styles.price}>
+            $
+            {price}
+          </Text>
+          <Text style={styles.quantity}>
+            {' '}
```
Comment (hwang635) [inline src/Marketplace/ProduceCard.js]: Replace the space with more margin/padding?

File: src/Marketplace/ProduceDetailsScreen.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+const styles = StyleSheet.create({
+  image: {
+    width: 205,
+    height: 386
+  }
+})
+
+function ProduceDetailsScreen({ route }) {
+  const { image, name, price } = route.params; // need help figuring out how to get the prop types
```
Comment (hwang635) [inline src/Marketplace/ProduceDetailsScreen.js]: label comments like these as TODO: [thing to do] for future ref. for prop types, we can go with the general field type in airtable like number/string/function/etc

File: src/Marketplace/MarketplaceScreen.js [hunk 6]
```diff
@@ -32,11 +46,51 @@ const styles = StyleSheet.create({
...
+      });
+      setProduceList(list);
+    });
+  };
+
+  const produceCards = produceList.map((produce) => (
+    <ProduceCard
+      style={styles.produceCard}
+      key={produce.Name}
+      navigation={navigation}
```
Comment (hwang635) [inline src/Marketplace/MarketplaceScreen.js:105]: Can you add checks for null values or some default value for the fields pulled from Airtable?

File: src/Marketplace/ProduceCard.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+            {unit}
+          </Text>
+        </View>
+      </View>
+    </TouchableOpacity>
+  );
+}
```
Comment (hwang635) [inline src/Marketplace/ProduceCard.js:97]: Could you add a little more space between the bottom of the image and start of the text?

**Review by hwang635 — CHANGES_REQUESTED** [review]
This looks really great! The produce card looks amazing and works really well. Only thing is that I don't think the heart thing does anything right now -- could you add a filled/not filled toggle functionality for the heart if pressed? I left a few questions and comments but otherwise this is perfect.

File: src/Marketplace/MarketplaceScreen.js (hunk 2 above)
Comment (he-is-james) [reply src/Marketplace/MarketplaceScreen.js:115]: Unsure, I assume it's how I set up pushing the data to the useState variables

**Review by hwang635 — APPROVED** [review]
Thanks for your hard work James!

**Pull request merged**